### DISCUSSION TITLE
Invalid URL

### DISCUSSION THREAD

**Discussion by seek-hope** [encode/httpx#3248] [Q&A]
When I use gradio and run relative python script, error: "httpx.InvalidURL: Invalid non-printable ASCII character in URL" occurs. Currently I simply comment out line 161 and 162 in _urlparse.py to fix this problem. Are there better ways to solve it?

**Comment by lovelydinosaur** [comment]
I guess the most common case where this occurs would be a trailing newline at the end of the URL.
Perhaps we could supplement the error message with a little more info, eg...

```
httpx.InvalidURL: Invalid non-printable ASCII character in URL. Character '\n' at position 123.
```

**Update**... addressed this in https://github.com/encode/httpx/pull/3250
